lib/ProductOpener/Data.pm

```diff
@@ -56,6 +67,22 @@ use Action::Retry;
 my $client;
 my $action = Action::CircuitBreaker->new();
 
+=head1 FUNCTIONS
+
+=head2 execute_query()
+
+C<execute_query()> executes a query on the database.
+
+=head3 Arguments
+
+Takes in a subroutine with a return type T.
+
+=head3 Return values
+
+The return value is of type T where T is the return type of the passed subroutine.
```

Comment:
I think we can indicate that the return value is the value returned by the query subroutine passed in parameter.

e.g. for a count,  we will get the count back:

```
$count = execute_query(sub {
			return get_products_collection()->count_documents($query_ref);
		});
```